File: Array/ThreeSum/Solution1.js
const threeSum=function(nums) {
    nums.sort((x,y)=>x-y)
    let n = arr.length;
    let newArr = []
    let a=0;
    let b=a+1;
    let c=b+1;
    let d =0;
    while(a<n-2) {
        while(c<arr.length) {

            if(arr[a] + arr[b] + arr[c] == 0){
                newArr[d]=[nums[a], nums[b], nums[c]]
                d++
            }
            c++
        }
        c=b+1;
        a++; b++; c++;
    }

    for(let i=0; i<newArr.length; i++) {
        for(let j=i+1; j<newArr.length; j++) {
            if(newArr[i]===newArr[j]) {
                delete newArr[j]
            }
        }
    }
    return newArr
}

let arr = [0,0,0,0]

console.log(threeSum(arr))

File: DSA/DataStructures/Array/10. left_P_Right/Solution1.java
public class Solution1 {

    public static void arrange(int[] arr,int p) {

        int n= arr.length;
        int[] newArr = new int[n];
        int i = 0;
        n--;
        int j = 0;
        int k = n;

        while(i<=n) {

            if(arr[i]<p) {
                newArr[j] = arr[i];
                j++;
            } else {
                newArr[k] = arr[i];
                k--;
            }
            i++;
        }

        for(int a : newArr) {
            System.out.print(a + " ");
        }
        System.out.println();
    }

    public static void main(String args[]) {
        int[] nums = {4,2,8,3,9,1,0,6};

        arrange(nums,4)  ;
    }
}